### PR TITLE
feat(GUI): enable test results pane - INS-4095

### PR DESCRIPTION
**Pull request by ihexxa**
### Changes
- [x] Merge to feat/test-result-pane
- [x] Added a tab for display test results
  - [x] Result list
  - [x] Button filter
  - [x] Fuzzy search filter
  - [x] Test counts
- [x] Minor style improvements according to requirements
- [x] Bring back tests

image

Ref: INS-4166

### CODE REVIEW

File: packages/insomnia/src/ui/components/panes/response-pane.tsx [hunk 1]
```diff
@@ -143,6 +144,19 @@ export const ResponsePane: FC<Props> = ({
...
   const cookieHeaders = getSetCookieHeaders(activeResponse.headers);
+
+  let passedTestCount = 0;
+  let totalTestCount = 0;
+  activeResponse.requestTestResults.forEach(result => {
+    if (result.status === 'passed') {
+      passedTestCount++;
+    }
+    totalTestCount++;
+  });
```
Comment (CurryYangxx) [inline packages/insomnia/src/ui/components/panes/response-pane.tsx]: Does this need to be put in useMemo？

**Review by filfreire — APPROVED** [review]

File: packages/insomnia/src/ui/components/panes/request-test-result-pane.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+            <div className="leading-4 m-auto mx-1">
+              <span className="mr-2 ml-2" >{statusTag}</span>
+            </div>
+            <div className="leading-4 mr-2">
+              <div className='mr-2 my-1 w-auto text-nowrap'>{message}</div>
+              <div className='text-sm text-neutral-400 my-1'>{`${testCategory} (`}{executionTime}{' ms)'}</div>
+            </div>
+          </div>
+        </div>);
+    });
```
Comment (CurryYangxx) [inline packages/insomnia/src/ui/components/panes/request-test-result-pane.tsx:89]: This approach to generate jsx element is too long and a bit difficult to read. I suggest abstracting into a separate component, or split data processing and jsx render.

**Review by jackkav — COMMENTED** [review]
i notice we allow tests in the pre request phase is this intentional?

**Comment by ihexxa** [comment]
> i notice we allow tests in the pre request phase is this intentional?

@jackkav Good catch, it is by intention, as it is also supported in other products.

File: packages/insomnia/src/ui/components/panes/request-test-result-pane.tsx (hunk 2 above)
Comment (ihexxa) [reply packages/insomnia/src/ui/components/panes/request-test-result-pane.tsx:89]: Good idea and moved to a separated component.

**Review by CurryYangxx — APPROVED** [review]

**Pull request merged**